ovs/lib/helpers/decorators.py

```diff
@@ -285,7 +288,7 @@ def update_value(key, append, value_to_update=None):
                 # Poll the arakoon to see whether this call is the first in list, if so --> execute, else wait
                 first_element = None
                 counter = 0
-                while first_element != now and counter < timeout:
+                while counter < timeout:
                     if persistent_client.exists(persistent_key):
                         value = persistent_client.get(persistent_key)
                         first_element = value['values'][0]['timestamp']
```

Comment:
Let's catch the index error here (which can be caused by the race condition we discussed) and make sure it eventually raises the ```EnsureSingleTimeoutReached``` exception, which is more correct. To be determined if we need a retry mechanism in the future or not.